bigquery/google/cloud/bigquery/client.py

```diff
@@ -1929,6 +1935,60 @@ def list_rows(
         )
         return row_iterator
 
+    def schema_from_json(self, file_or_path):
+        """Takes a file object or file path that contains json that describes
+            a table schema.
+
+            Returns:
+                List of schema field objects.
+        """
+        file_obj = None
+        json_data = None
+        schema_field_list = list()
+
+        if isinstance(file_or_path, io.IOBase):
+            file_obj = file_or_path
+        else:
+            try:
+                file_obj = open(file_or_path)
+            except OSError:
+                raise TypeError(_NEED_JSON_FILE_ARGUMENT)
+
+        try:
+            json_data = json.load(file_obj)
+        except JSONDecodeError:
+            raise TypeError(_NEED_JSON_FILE_ARGUMENT)
+
+        for field in json_data:
+            schema_field = SchemaField.from_api_repr(field)
+            schema_field_list.append(schema_field)
+
+        return schema_field_list
+
+    def schema_to_json(self, schema_list, destination):
+        """Takes a list of schema field objects.
+
+            Serializes the list of schema field objects as json to a file.
+
+            Destination is a file path or a file object. 
+        """
+        file_obj = None
+        json_schema_list = list()
+
+        if isinstance(destination, io.IOBase):
+            file_obj = destination
+        else:
+            try:
+                file_obj = open(destination, mode="w")
```

Comment:
Just as in `schema_from_json`, we *must close* `file_obj` if we open it based on a path, but we *must **not** close* `file_obj` if we are given a file-like object.

Reply:
Updated with `with` statement to close when appropriate.